### ISSUE TITLE
Infinite loop when walking through archive containing . (dot) entry

### ISSUE THREAD

**Issue by adyatlov** [nlepage/go-tarfs#10]
STEPS
1. Create an archive:

    ```bash
    mkdir empty_dir
    tar -cf fs_test_infinite_loop.tar -C empty_dir .
    ```
    
2. Walk the archive using `fs.WalkDir` function.

EXPECTED:
The walk function is called once.
OBSERVED:
The walk function is in an infinite loop.

**Comment by nlepage** [comment]
fixed by #11

**Comment by nlepage** [comment]
released v1.0.6 https://github.com/nlepage/go-tarfs/releases
